src/hera/shared/_type_util.py

```diff
@@ -127,6 +134,22 @@ def get_unsubscripted_type(t: Any) -> Any:
     return t
 
 
+def is_optionally_subtype(annotation: Any, type_: Union[type, Tuple[type, ...]]) -> bool:
+    """Return True if annotation is a subtype of type_ or None.
+
+    This is useful for checking if an annotation is an Optional[type].
+    """
+    unwrapped_type = unwrap_annotation(annotation)
+    origin_type = get_unsubscripted_type(unwrapped_type)
+    if origin_type is Union or origin_type is UnionType:
+        if len(get_args(unwrapped_type)) == 2 and type(None) in get_args(unwrapped_type):
+            return any(origin_type_issubtype(arg, type_) for arg in get_args(unwrapped_type))
+    else:
+        return isinstance(origin_type, type) and issubclass(origin_type, type_)
+
+    return False
```

Comment:
Is this different from calling `origin_type_issubtype(annotation, (type_, type(None)))`?

Reply:
This function was unnecessary in the end so have removed. It is indeed equivalent to `origin_type_issubtype(annotation, (type_, type(None)))` though 👍